src/WrappedMToken.sol

```diff
@@ -521,16 +538,16 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
     function _subtractTotalEarningSupply(uint240 amount_, uint128 currentIndex_) internal {
         if (amount_ >= totalEarningSupply) {
             totalEarningSupply = 0;
-            _principalOfTotalEarningSupply = 0;
+            principalOfTotalEarningSupply = 0;
 
             return;
         }
 
         unchecked {
             uint112 principal_ = IndexingMath.getPrincipalAmountRoundedDown(amount_, currentIndex_);
 
-            _principalOfTotalEarningSupply -= (
-                principal_ > _principalOfTotalEarningSupply ? _principalOfTotalEarningSupply : principal_
+            principalOfTotalEarningSupply -= (
```

Comment:
for round down it should be safe to do 

` _principalOfTotalEarningSupply -= principal`

but let's fix it in future versions